src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,380 @@
+     * Instances are fetched from API by skus, that corresponds to skus present in {@code drafts}.
+     *
+     * @param drafts {@link List} of drafts
+     * @return {@link SkuKeyTuple} to {@link InventoryEntry} of existing entries that correspond to passed
+     * {@code drafts} by sku comparision.
+     */
+    private Map<SkuKeyTuple, InventoryEntry> fetchExistingInventories(final List<InventoryEntryDraft> drafts) {
+        final Set<String> skus = extractSkus(drafts);
+        final List<InventoryEntry> existingEntries = inventoryService.fetchInventoryEntriesBySkus(skus);
+        return mapSkuAndKeyToInventoryEntry(existingEntries);
+    }
+
+    /**
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} where each draft contains its sku
+     * @return {@link Set} of distinct skus found in {@code inventories}
+     */
+    private Set<String> extractSkus(final List<InventoryEntryDraft> inventories) {
+        return inventories.stream()
+                .map(InventoryEntryDraft::getSku)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Creates map of {@link SkuKeyTuple} to {@link InventoryEntry}, so that all distinct tuples of sku and supply
+     * channel key from {@code inventories} are mapped to entry instance. For use in comparision with
+     * {@link InventoryEntryDraft} i.e. create mapping from existing entries and then decide if specific draft
+     * should be used for creation of new entry or for building update actions for existing entry (by checking if
+     * sku-key tuple of draft is present in returned mapping)
+     *
+     * @param inventories list of {@link InventoryEntry}
+     * @return {@link Map} of {@link SkuKeyTuple} to {@link InventoryEntry}
+     */
+    private Map<SkuKeyTuple, InventoryEntry> mapSkuAndKeyToInventoryEntry(final List<InventoryEntry> inventories) {
+        final Map<SkuKeyTuple, InventoryEntry> skuKeyToInventory = new HashMap<>();
+        inventories.forEach(entry -> skuKeyToInventory.put(SkuKeyTuple.of(entry), entry));
+        return skuKeyToInventory;
+    }
+
+    /**
+     * Tries to update {@code entry} in database with data from {@code draft}.
+     * It calculates list of {@link UpdateAction} and calls API only when there is a need.
+     * Before calculate differences, the channel reference from {@code draft} is replaced, so it points to
+     * proper channel ID in target system.
+     * It either updates inventory entry and increments "updated" counter in statistics, or increments "failed" counter
+     * and log error in case of any exception.
+     *
+     * @param entry entry from existing system that could be updated.
+     * @param draft draft containing data that could differ from data in {@code entry}.
+     *              <strong>Sku isn't compared</strong>
+     */
+    private void attemptUpdate(final InventoryEntry entry, final InventoryEntryDraft draft) {
```

Comment:
It's only single-threaded processing now 100d96ff515d13f645bc086a9c10fcb46881499d